language/vm/src/file_format.rs

```diff
@@ -1224,6 +1224,55 @@ pub enum Bytecode {
     ///
     /// ```..., u64_value(1), u64_value(2) -> ..., u64_value```
     Shr,
+    /// Create an empty vector.
+    ///
+    /// Stack transition:
+    ///
+    /// ```... -> ..., vec[]```
+    VecEmpty(SignatureIndex),
+    /// Return the length of the vector,
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., vector_reference -> ..., u64_value```
+    VecLen,
+    /// Acquire an immutable reference to the element at a given index of the vector. Abort the
+    /// execution if the index is out of bounds.
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., vector_reference, u64_value -> .., element_reference```
+    VecImmBorrow,
+    /// Acquire a mutable reference to the element at a given index of the vector. Abort the
+    /// execution if the index is out of bounds.
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., vector_reference, u64_value -> .., element_reference```
+    VecMutBorrow,
+    /// Add an element to the end of the vector.
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., vector_reference, element -> ...```
+    VecPushBack,
+    /// Pop an element from the end of vector. Aborts if the vector is empty.
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., vector_reference -> ..., element```
+    VecPopBack,
+    /// Destroy the vector. Aborts if the vector is not empty
+    ///
+    /// Stack transition:
+    ///
+    /// ```..., vec[] -> ...`
+    VecDestroy,
```

Comment:
I think we should name this `VecDestroyEmpty`

Reply:
Shall we do `VecEmpty` -> `VecCreateEmpty`?